src/main/java/org/schemaspy/progress/Progress.java

```diff
@@ -0,0 +1,12 @@
+package org.schemaspy.progress;
+
+/**
+ * Abstraction to allow a {@link Job} to track progression
```

Comment:
Since `Job` knows about `Progress`, `Progress` should not know about `Job`.
The concept is not independent, so perhaps it's better of as a decorator for `Job`?

Reply:
We are mid refactoring I hope to reach this intermediate state where we can first decouple "stages" later we can think about jobs and tasks.

As I mentioned above. I want the ability to decouple stages and eventually tasks. In short with this intermediary state we can remove the ProgresdListener an maybe think more about independent things that need to be "done" and that they "need" to be tracked.

In the long run we should be able to declare things that should be done and if we choose to track progress or not is a decorator for a task.